x-pack/plugins/lens/public/lens_ui_errors/error_boundary.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiErrorBoundary } from '@elastic/eui';
+
+interface ErrorBoundaryProps {
+  children: JSX.Element;
+  onError?: (error: Error) => void;
+}
+
+interface ErrorBoundaryState {
+  originalError?: Error;
+}
+
+/** @internal **/
+const RecallError = ({ error }: { error: Error }) => {
+  throw error;
```

Comment:
It's handling the case but should we open an issue for EUI so they expose `onError` natively?

Reply:
https://github.com/elastic/eui/issues/6127